crates/parry3d/examples/plane_intersection.rs

```diff
@@ -0,0 +1,156 @@
+use macroquad::models::Vertex;
+use macroquad::prelude::*;
+use nalgebra::{Point3, UnitVector3, Vector3};
+use parry3d::math::Real;
+use parry3d::query::IntersectResult;
+use parry3d::shape::{Cuboid, TriMesh};
+
+#[macroquad::main("parry3d::query::PlaneIntersection")]
+async fn main() {
+    let trimesh = Cuboid::new(Vector3::new(1.0, 1.0, 1.0)).to_trimesh();
+
+    let camera_pos = Vec3::new(-1.5f32, 2.5f32, -3f32);
+
+    let mesh = mquad_mesh_from_points(&trimesh, camera_pos);
+    let trimesh = TriMesh::new(trimesh.0, trimesh.1);
+
+    for _ in 1.. {
+        clear_background(BLACK);
+
+        let elapsed_time = get_time();
+
+        // Animated rotation for the intersection plane.
+        let bias = -1.2 * (elapsed_time as f32 / 3f32).sin();
+        let rotation = Quat::from_axis_angle(Vec3::Z, (elapsed_time as f32 * 40f32).to_radians());
+        let up_plane_vector = rotation * Vec3::Y;
+
+        // Get the intersection polyline.
+        let intersection_result = trimesh.intersection_with_local_plane(
+            &UnitVector3::new_normalize(Vector3::<Real>::new(
```

Comment:
Is the type parameter annotation really necessary?
```suggestion
            &UnitVector3::new_normalize(Vector3::new(
```